src/resources/dart-swift-concurrency.md

```diff
@@ -0,0 +1,303 @@
+```dart
+class HomePage extends StatelessWidget {
+  final HomePageViewModel viewModel = const HomePageViewModel();
+  const HomePage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      // Feed a FutureBuilder to your widget tree.
+      child: FutureBuilder<Weather>(
+        // Specify the Future that you want to track.
+        future: viewModel.load(),
+        builder: (context, snapshot) {
+          // A snapshot is of type `AsyncSnapshot` and contains the
+          // state of the Future. By looking if the snapshot contains
+          // an error or if the data is null, you can decide what to
+          // show to the user.
+          if (snapshot.hasData) {
+            return Center(
+              child: Text(
+                snapshot.data.toString(),
+              ),
+            );
+          } else {
+            return const Center(
+              child: CupertinoActivityIndicator(),
+            );
+          }
+        },
+      ),
+    );
+  }
+}
+```
+
+You can see the complete (example on Github)[https://github.com/flutter/website/tree/main/examples/resources/lib/async_weather.dart].
+
+### Leveraging Background Threads or Isolates
+Flutter apps can run on a variety of multi-core hardware, 
+including devices running macOS and iOS. 
+To improve the performance of these applications, 
+it’s sometimes necessary to run tasks 
+on the different cores concurrently- 
+especially to avoid blocking UI rendering 
+on the main thread with long running operations. 
+
+In Swift, you can leverage GCD to run tasks on global queues
+ with different quality of service class 
+ (qos) properties- which indicate the tasks’ priority. 
+
+```swift
+func parse(string: String, completion: @escaping ([String:Any]) -> Void) {
+  // Mimic 1 sec delay.
+  DispatchQueue(label: "data_processing_queue", qos: .userInitiated)
+    .asyncAfter(deadline: .now() + 1) {
+      let result: [String:Any] = ["foo": 123]
+      completion(result)
+    }
+  }
+}
+```
+
+In Dart, you can offload computation to a worker isolate, 
+often called a background worker. 
+A common case is spawning a simple worker isolate and 
+returning the results in a message when the worker exits. 
+As of Dart 2.19, `Isolate.run()` can be used to 
+spawn an isolate and run some computation:
+
+```dart
+void main() async {
+  // Read some data.
+  final jsonData = await Isolate.run(() => jsonDecode(jsonString) as Map<String, dynamic>);`
+
+  // Use that data.
+  print('Number of JSON keys: ${jsonData.length}');
+}
+```
+
+In Flutter, you are also able to use the compute function 
+to spin up an isolate to run a callback function:
+
+```dart
+final jsonData = await compute(getNumberOfKeys, jsonString);
+
+In this case, the call back function is a top-level function as shown below:
+Map<String, dynamic> getNumberOfKeys(String jsonString) {
+ return jsonDecode(jsonString);
+}
+```
```

Comment:
This page ends kind of abruptly. Can you add some brief text here? Perhaps to link them to more information?